pvlib/iotools/maccrad.py

```diff
@@ -0,0 +1,155 @@
+# standard library imports 
+import logging
+
+# related third party imports
+import pandas as pd
+import pytz
+
+# local application/library specific imports 
+from pvlib.location import Location
+from pvlib.iotools.util import get_loc_latlon, localise_df
+
+# required dateconverters
+def dtp_soda_pro_macc_rad(date):
+    """
+    datetime converter for MACC-RAD data and others
+    """
+    datetime_stamp = date.split('/')[0]
+
+    return datetime_stamp
+    
+#XXX read metadata / header
+
+def read_maccrad_metadata(file_csv, name='maccrad'):
+    """
+    Read metadata from commented lines of the file
+    * coordinates
+    * altitude
+    
+    Retrieve timezone
+    """
+    if not name:
+        name = file_csv.split('.')[0]
+    
+    ## if file is on local drive
+    f = open(file_csv)
+    for line in f:
+        if "Title" in line:
+            title = line.split(':')[1].split('(')[0].strip()
+            name = title
+        if "Latitude" in line:
+    #    print (line)
+    #    if line.startswith( "# Latitude"):
+            lat_line = line
+            lat = float(lat_line.split(':')[1])
+    #        lat = float(line.split(':')[1])
+        if "Longitude" in line:
+    #    if line.startswith( "# Longitude"):
+            lon_line = line
+            lon = float(lon_line.split(':')[1])
+            lon = float(line.split(':')[1])
+    #    if line.startswith( "# Altitude"):
+        if "Altitude" in line:
+            alt_line = line
+            alt = float(alt_line.split(':')[1])
+    #        alt = float(line.split(':')[1])
+        if "Time reference" in line:
+            if "Universal time (UT)" in line:
+                tz_raw = 'UTC'
+            else:
+                logging.debug('No metadata on timezone found in input file')
+                logging.debug('Assuming UTC as default timezone')
+                tz_raw = 'UTC'
+
+    tz_loc = get_loc_latlon(lat, lon)
+    
+    
+    
+    location = Location(lat, lon, name=name, altitude=alt,
```

Comment:
``Location`` is part of the higher level API, so we don't want to force people into it in a low level reader. Instead, build and return a dict. You can add a constructor method to ``Location`` that consumes that dict.